vms/platformvm/txs/txheap/heap.go

```diff
@@ -23,107 +20,57 @@ type Heap interface {
 }
 
 type heapTx struct {
-	tx    *txs.Tx
-	index int
-	age   int
+	tx  *txs.Tx
+	age int
 }
 
 type txHeap struct {
-	self heap.Interface
-
-	txIDToIndex map[ids.ID]int
-	txs         []*heapTx
-	currentAge  int
-}
-
-func (h *txHeap) initialize(self heap.Interface) {
-	h.self = self
-	h.txIDToIndex = make(map[ids.ID]int)
+	heap       heap.Map[ids.ID, heapTx]
+	currentAge int
 }
 
 func (h *txHeap) Add(tx *txs.Tx) {
-	heap.Push(h.self, tx)
+	txID := tx.ID()
+	if h.heap.Contains(txID) {
+		return
+	}
+	htx := heapTx{
+		tx:  tx,
+		age: h.currentAge,
+	}
+	h.currentAge++
+	h.heap.Push(txID, htx)
 }
 
 func (h *txHeap) Get(txID ids.ID) *txs.Tx {
-	index, exists := h.txIDToIndex[txID]
-	if !exists {
-		return nil
-	}
-	return h.txs[index].tx
+	got, _ := h.heap.Get(txID)
+	return got.tx
 }
 
 func (h *txHeap) List() []*txs.Tx {
-	res := make([]*txs.Tx, 0, len(h.txs))
-	for _, tx := range h.txs {
+	heapTxs := heap.MapValues(h.heap)
+	res := make([]*txs.Tx, 0, len(heapTxs))
+	for _, tx := range heapTxs {
 		res = append(res, tx.tx)
 	}
 	return res
 }
 
 func (h *txHeap) Remove(txID ids.ID) *txs.Tx {
-	index, exists := h.txIDToIndex[txID]
-	if !exists {
-		return nil
-	}
-	return heap.Remove(h.self, index).(*txs.Tx)
+	removed, _ := h.heap.Remove(txID)
```

Comment:
I don't think `removed` is guaranteed to be non-nil?

Reply:
`removed ` is a struct, not a pointer. So its `zero` value will produce the expected result here.